common/sgx/rand.S

```diff
@@ -3,15 +3,28 @@
 
 .text
 .globl  oe_rdrand
+.type oe_rdrand, @function
```

Comment:
Consider replacing this external assembler code with GCC/clang inline assembly in Linux.
It will allow the compiler to perform better optimizations and would lead to code as close to performance as the GCC intrinsics themselves.

```c
OE_INLINE uint64_t oe_rdrand()
{
    uint64_t rv = 0;
    asm volatile(
	"_rdrand_retry: \n"
        "    rdrand %0 \n"
	"    jnc _rdrand_retry \n"
	: "=a"(rv)  // Outputs to rax 
	:           // No inputs
	: "cc"); // Clobbers flags
    return rv;
}
```

Reply:
I would not optimize this specifically for `oe_rdrand()` since:

- The syntax for inline assembly has not been well-handled in the history of this project (e.g. #144, #189) In the area of security features, I would prioritize readability and maintainability relative to performance.

- There's an established pattern of splitting Linux/Windows implementations as .S/.asm pairs. If you think there's a valuable optimization that should be generally applied for how we deal with Linux assembly, (e.g. .c/.asm pairs) I think that should be a separate refactor, and we should establish that as a standard pattern of use with the committers.